src/condor_contrib/docker/README.md

```diff
@@ -0,0 +1,145 @@
+
+
+### Example
+
+This example is for creating an execute node and adding it to a pool using
+token auth.  The execute node will have 2 CPUs and 4 GB of RAM, and will
+use the identity `dockerworker@example.net`.  The central manager has
+the hostname `cm.example.net`.  The container is run by the user `user`
+on the host `dockerhost.example.net`.
+
+Create a directory for holding the token:
+```console
+dockerhost$ mkdir -p ~/condorexec/secrets
+dockerhost$ chmod 0700 ~/condorexec/secrets
+```
+
+Grant access to the identity that the container will use.  On the central
+manager, add the following lines to the HTCondor configuration:
+```
+ALLOW_ADVERTISE_MASTER = $(ALLOW_ADVERTISE_MASTER) dockerworker@example.net
+ALLOW_ADVERTISE_STARTD = $(ALLOW_ADVERTISE_STARTD) dockerworker@example.net
+```
+Run `condor_reconfig` on the central manager to pick up the changes.
+
+Create a token for the execute node.  On the central manager:
+```console
+cm$ sudo condor_create_token -authz ADVERTISE_MASTER \
+         -authz ADVERTISE_STARTD -identity dockerworker@example.net \
+         dockerworker_token
+cm$ sudo scp user@dockerhost.example.net:volumes/condorexec/secrets/token
+```
+
+On the Docker host, create an environment file for the container:
+```console
+dockerhost$ echo 'CONDOR_HOST=cm.example.net' > ~/condorexec/env
+dockerhost$ echo 'NUM_CPUS=2' >> ~/condorexec/env
+dockerhost$ echo 'MEMORY=4096' >> ~/condorexec/env
+```
+
+Start the container:
+```console
+dockerhost$ docker run --env-file=~/condorexec/env \
+                -v ~/condorexec/secrets:/root/secrets:ro \
+                htcondor/execute:el7
+```
+
+To verify the container is functioning, use `docker ps` to get the name
+of the container, then run:
+```console
+dockerhost$ docker exec -ti <container name> /bin/bash
+container$ condor_status
+```
+You should see the container in the output of `condor_status`.
+
+
+
+Building
+--------
+
+The images are built using a Makefile.  The Makefile needs to know what
+version of HTCondor the images are built for, in order to tag them properly.
+The images for the various roles (`execute`, `cm`, `mini`, `submit`)
+are built on top of the `htcondor/base` image, which gets built first.
+`htcondor/base` contains the HTCondor software, some common configuration
+and scripts.
+
+To build all the images, run:
+```console
+$ make build VERSION=8.9.5
+```
+
+The images will be named like `htcondor/execute:8.9.5-el7`.  To create
+versionless aliases for the built products (e.g. `htcondor/execute:el7`),
+pass `UNVERSIONED=yes` to `make`.
+
+
+To push the images to Docker Hub, run:
+```console
+$ make push VERSION=8.9.5
+```
+
+To push to an alternate registry, set the `REGISTRY` parameter to the
+registry you want to push to:
+```console
+$ make push VERSION=8.9.5 REGISTRY=registry.example.net
+```
+
+To push the versionless tags for the images, pass `UNVERSIONED=yes` to
+`make`.
+
+See `make help` for additional targets and options.
+
```

Comment:
Should there also be a section on customizing images, e.g. how to add your own software to an execute node?